src/ui-test/tests/code.navigation.test.ts

```diff
@@ -0,0 +1,229 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assert } from "chai";
+import { ActivityBar, DefaultTreeSection, InputBox, QuickPickItem, SideBarView, VSBrowser, WebDriver, Workbench } from "vscode-uitests-tooling";
+import { CODE_NAVIGATION_XML, CODE_NAVIGATION_JAVA, RESOURCES, closeEditor, waitUntilEditorIsOpened, waitUntilExtensionIsActivated, activateEditor } from "../utils/testUtils";
+import * as pjson from '../../../package.json';
+import * as path from 'path';
+
+let driver: WebDriver;
+
+describe('Code navigation', function () {
+    this.timeout(90000);
+
+    let input: InputBox;
+    let quickpicks: QuickPickItem[];
+    let section: DefaultTreeSection;
+
+    before(async function () {
+        this.timeout(40000);
+
+        driver = VSBrowser.instance.driver;
+
+        await VSBrowser.instance.openResources(RESOURCES);
+        await VSBrowser.instance.waitForWorkbench();
+
+        await waitUntilExtensionIsActivated(driver, `${pjson.displayName}`);
+
+        await (await new ActivityBar().getViewControl('Explorer')).openView();
+        await (await new SideBarView().getContent().getSection('resources')).collapse();
+
+        section = await new SideBarView().getContent().getSection('Outline') as DefaultTreeSection;
+        await section.expand();
+    });
+
+    describe('XML DSL', function () {
+
+        // All available symbols in 'camel-route.xml' with line of occurence.
+        const XML_av_symbols = [
+            ['uitest-context', 1],
+            ['uitest-direct-route', 2],
+            ['from direct:testName1', 3],
+            ['to direct:testName2', 4],
+            ['uitest-directvm-route', 7],
+            ['from direct-vm:testName1', 8],
+            ['to direct-vm:testName2', 9]
+        ];
+
+        beforeEach(async function () {
+            await VSBrowser.instance.openResources(path.join(RESOURCES, CODE_NAVIGATION_XML));
+            await waitUntilEditorIsOpened(driver, CODE_NAVIGATION_XML);
+        });
+
+        afterEach(async function () {
+            await closeEditor(CODE_NAVIGATION_XML, false);
+        });
+
+        describe('Quickpick \'Go to Symbol in Editor\' navigation', function () {
```

Comment:
is this different test or we can unify naming of this and following describe with Java DSL ones?